ouroboros-consensus-test/src/Test/Util/Orphans/DiffSeq/Arbitrary.hs

```diff
@@ -0,0 +1,44 @@
+{-# LANGUAGE DerivingStrategies         #-}
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Test.Util.Orphans.DiffSeq.Arbitrary () where
+
+import           Test.QuickCheck hiding (Fixed (..))
+
+import qualified Data.Map.Diff.Strict as MapDiff
+import qualified Ouroboros.Consensus.Storage.LedgerDB.HD.DiffSeq as DS
+import qualified Ouroboros.Consensus.Storage.LedgerDB.HD.TableTypes as TT
+
+import           Test.Util.Orphans.Slotting.Arbitrary ()
+
+{------------------------------------------------------------------------------
+  Diffs
+------------------------------------------------------------------------------}
+
+deriving newtype instance (Ord k, Arbitrary k, Arbitrary v)
+                       => Arbitrary (MapDiff.Diff k v)
+deriving newtype instance (Arbitrary v) => Arbitrary (MapDiff.DiffHistory v)
+instance (Arbitrary v) => Arbitrary (MapDiff.DiffEntry v) where
+  arbitrary = do
+    constr <- elements [MapDiff.Insert, MapDiff.Delete]
+    constr <$> arbitrary
```

Comment:
This is not the most general arbitrary instance. Instead, create a `newtype` wrapper that has this instance's behaviour.